### PR TITLE
Travis

### PR DESCRIPTION
**Pull request by bnkamalesh**
- Minor cleanup
- Added tests
- added config for travis

ref #11

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,12 +24,47 @@ import (
...
+		// img.shields.io response time is very slow
+		Timeout: 30 * time.Second,
+	}
+
+	// ErrImgUnSupported is the error returned when the Go version requested is
+	// not in the supported list
+	ErrImgUnSupported = errors.New("Unsupported Go version provided")
+)
+
+var (
```
Comment (avelino) [inline main.go]: Group `var`s declarete

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		t.Log(err)
+		t.Fail()
+	}
+	if string(bytes) != expected {
+		t.Log("Expected svg badge, got", string(bytes))
+		t.Fail()
+	}
+}
+
+func TestRun(t *testing.T) {
```
Comment (avelino) [inline main_test.go:41]: Create negative case, where you receive a tag that does not exist

What `run()` behavior?

**Review by avelino — CHANGES_REQUESTED** [review]

File: main_test.go (hunk 2 above)
Comment (bnkamalesh) [reply main_test.go:41]: [](https://travis-ci.org/bnkamalesh/cover.run)

Negative tests added.

File: main_test.go [hunk 3]
```diff
@@ -58,16 +70,28 @@ func TestRepoCover(t *testing.T) {
...
+	}
+
+	_, err = repoCover("github.com/avelino/nonexistent", "golang-1.10")
+	if err.Error() != "provision: container exited with failure" {
+		t.Log("Expected 'provision: container exited with failure'. got", err)
 		t.Fail()
 	}
 }
 
+// func TestRepoLatest(t *testing.T) {
```
Comment (avelino) [inline main_test.go]: removed comment source

**Review by avelino — CHANGES_REQUESTED** [review]

**Review by avelino — APPROVED** [review]
LGTM

**Comment by bnkamalesh** [comment]
@avelino, silly me forgot to add the Travis badge on README!

**Pull request merged**